### PR TITLE
Add CRD for actiongroups

### LINKED ISSUE #485: CRD for Action Groups
Following up on #254 , this issue is track adding CRD for the action_groups.yml in the securityconfig-secret. Similar to existing CRDs like opensearchusers, opensearchroles etc. a new CRD for action groups i.e. opensearchactiongroups will be added. 

The proposed API:

```
apiVersion: opensearch.opster.io/v1
kind: OpensearchActionGroup
metadata:
  name: sample-action-group
  namespace: default
spec:
  opensearchCluster:
    name: my-first-cluster
  allowedActions:                                #type []string
    - "indices:admin/aliases/get",
    - "indices:admin/aliases/exists"
  type: "index"                                  #type string
  description: "Sample action group"             #type string
```

These fields were decided based on the OS API and what it supports for the action groups via REST API. [Reference](https://opensearch.org/docs/2.6/security/access-control/api/#action-groups).

### PR DESCRIPTION
**Pull request by saketmht**
This fixes #485 

Changes:

- Add new CRD to manage actiongroups in opensearch
- Added code for the controller to watch and reconcile the CRD
- Added tests for the controller code
- Some refactoring of existing code
- Doc changes

### CODE REVIEW

**Comment by saketmht** [comment]
@swoehrl-mw @idanl21 Could you please review? 

PS - The change in the OS Cluster CRD is due to the recent PR #476 which added new fields to the CRD but didn't update the CRD using `make manifests`.

**Comment by saketmht** [comment]
@swoehrl-mw Could you please review?

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
Code looks good to me. One minor thing. Once that is changed we can merge.

File: docs/developing.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ All PRs must conform to the following rules:
         cp opensearch-operator/config/crd/bases/opensearch.opster.io_opensearchroles.yaml charts/opensearch-operator/templates/OpensearchRole-crd.yaml
         cp opensearch-operator/config/crd/bases/opensearch.opster.io_opensearchuserrolebindings.yaml charts/opensearch-operator/templates/OpensearchUserRoleBinding-crd.yaml
         cp opensearch-operator/config/crd/bases/opensearch.opster.io_opensearchusers.yaml charts/opensearch-operator/templates/OpensearchUser-crd.yaml
+        cp opensearch-operator/config/crd/bases/opensearch.opster.io_opensearchactiongroups.yaml charts/opensearch-operator/templates/OpensearchActionGroup-crd.yaml
```
[the remark this answers is not in the document]
Comment (saketmht) [reply docs/developing.md:122]: updated the release workflow file

**Review by swoehrl-mw — APPROVED** [review]

**Pull request merged**